resin/models/application.py

```diff
@@ -217,27 +221,25 @@ def restart(self, name):
             endpoint=self.settings.get('api_endpoint'), login=True
         )
 
-    def get_api_key(self, name):
+    def get_config(self, app_id):
         """
-        Get the API key for a specific application. This function only works if you log in using credentials or Auth Token.
+        Download application config.json.
 
         Args:
             name (str): application name.
 
         Returns:
-            str: API key.
+            dict: application config.json content.
 
         Raises:
-            ApplicationNotFound: if application couldn't be found.
+            Request error: if user doesn't have permission to access the application.
```

Comment:
Not sure if this is controlled by the SDK, but it's usually better to obscure the fact that something is inaccessible by someone by raising a NotFound exception. Although if that's dictated by the API itself, I suggest not changing this.